packages/react-compose/src/next/getSlots.test.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { getSlots } from './getSlots';
+import { nullRender } from './nullRender';
+
+describe('getSlots', () => {
+  const Foo = () => <div />;
+
+  it('renders nullRender for root if the as prop is not provided', () => {
+    expect(getSlots({})).toEqual({
+      slots: { root: nullRender },
+      slotProps: { root: {} },
+    });
+  });
+
+  it('renders root as a span with no props', () => {
+    expect(getSlots({ as: 'span' })).toEqual({
+      slots: { root: 'span' },
+      slotProps: { root: {} },
+    });
+  });
+
+  it('renders root as a button, omitting the href prop', () => {
+    expect(getSlots({ as: 'button', id: 'id', href: 'href' })).toEqual({
+      slots: { root: 'button' },
+      slotProps: { root: { id: 'id' } },
+    });
+  });
+
+  it('renders root as an anchor, leaving the href intact', () => {
+    expect(getSlots({ as: 'a', id: 'id', href: 'href' })).toEqual({
+      slots: { root: 'a' },
+      slotProps: { root: { id: 'id', href: 'href' } },
+    });
+  });
+
+  it('renders root as a component, leaving the props intact minus the as prop', () => {
+    expect(getSlots({ as: Foo, id: 'id', href: 'href', blah: 1 })).toEqual({
+      slots: { root: Foo },
+      slotProps: { root: { id: 'id', href: 'href', blah: 1 } },
+    });
+  });
+
+  it('renders null for primitive slots with no children', () => {
+    expect(getSlots({ as: 'div', icon: { as: 'span' } }, ['icon'])).toEqual({
+      slots: { root: 'div', icon: nullRender },
+      slotProps: { root: {} },
+    });
+  });
+
+  it('renders a component slot with no children', () => {
+    expect(getSlots({ as: 'div', icon: { as: Foo } }, ['icon'])).toEqual({
+      slots: { root: 'div', icon: Foo },
+      slotProps: { root: {}, icon: {} },
+    });
+  });
+
+  it('renders slot as button and omits unsupported props (href)', () => {
```

Comment:
I like this description on the unsupported props more 👍

Reply:
addressed most of the descriptions :)